worker/containerdworker/worker.go

```diff
@@ -53,26 +60,40 @@ func (w containerdWorker) Exec(ctx context.Context, meta worker.Meta, root cache
 	if err != nil {
 		return err
 	}
-	defer task.Delete(ctx)
+	defer func() {
+		if _, err1 := task.Delete(context.TODO()); err == nil && err1 != nil {
+			err = errors.Wrapf(err1, "failed to delete task %s", id)
+		}
+	}()
 
 	// TODO: Configure bridge networking
 
-	// TODO: support sending signals
-
 	if err := task.Start(ctx); err != nil {
 		return err
 	}
 
-	statusCh, err := task.Wait(ctx)
+	statusCh, err := task.Wait(context.Background())
 	if err != nil {
 		return err
 	}
-	status := <-statusCh
-	if status.ExitCode() != 0 {
-		return errors.Errorf("process returned non-zero exit code: %d", status.ExitCode())
+
+	killCtx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+
+	ctxDone := ctx.Done()
+	for {
+		select {
+		case <-ctxDone:
+			ctxDone = nil
```

Comment:
nit: why not just `case <-ctx.Done():`?

Reply:
In the second iteration, this case is invalid. I guess setting `ctx = context.Background()` would work as well, but maybe would look weirder.